solutions/usaco-894.mdx

```diff
@@ -19,222 +19,292 @@ It can be shown that the tree can also be colored accordingly in
+
+	vector<int> deg(field_num + 1);  // 1-indexed
+	for (int p = 0; p < field_num - 1; p++) {
+		int field1, field2;
+		cin >> field1 >> field2;
+
+		deg[field1]++;
+		deg[field2]++;
+	}
+
+	int max_deg = 0;
+	for (int f = 1; f <= field_num; f++) {
+		max_deg = max(max_deg, deg[f]);
 	}
-	cout << maxDeg + 1 << "\n";
+
+	cout << max_deg + 1 << endl;
 }
 ```
 
 </CPPSection>
-
 <JavaSection>
 
 ```java
 import java.io.*;
 import java.util.StringTokenizer;
 
-public class GrassPlanting {
+public class Planting {
 	public static void main(String[] args) throws IOException {
 		Kattio io = new Kattio("planting");
-		int n = io.nextInt();
-		int[] deg = new int[n+1];
-		for(int i = 0; i < n-1; i++) {
-			int a = io.nextInt(), b = io.nextInt();
-			deg[a]++; deg[b]++;
-		}
-
-		int ret = 0;
 
-		for(int i = 1; i <= n; i++) ret = Math.max(ret, deg[i]);
-		io.println(ret+1);
-		io.close();
-	}
-
-	private static class Kattio extends PrintWriter {
-		private BufferedReader r;
-		private StringTokenizer st;
-
-		// standard input
-		public Kattio() { this(System.in,System.out); }
-
-		public Kattio(InputStream i, OutputStream o) {
-			super(o);
-			r = new BufferedReader(new InputStreamReader(i));
+		int fieldNum = io.nextInt();
+		int[] deg = new int[fieldNum + 1];  // 1-indexed
+		for (int p = 0; p < fieldNum - 1; p++) {
+			int field1 = io.nextInt();
+			int field2 = io.nextInt();
+			deg[field1]++;
+			deg[field2]++;
 		}
 
-		// USACO-style file input
-		public Kattio(String problemName) throws IOException {
-			super(new FileWriter(problemName+".out"));
-			r = new BufferedReader(new FileReader(problemName+".in"));
+		int maxDeg = 0;
+		for(int f = 1; f <= fieldNum; f++) {
+			maxDeg = Math.max(maxDeg, deg[f]);
 		}
 
-		// returns null if no more input
-		public String next() {
-			try {
-				while (st == null || !st.hasMoreTokens())
-					st = new StringTokenizer(r.readLine());
-				return st.nextToken();
-			} catch (Exception e) {}
-			return null;
-		}
-		public int nextInt() { return Integer.parseInt(next()); }
-		public double nextDouble() { return Double.parseDouble(next()); }
-		public long nextLong() { return Long.parseLong(next()); }
+		io.println(maxDeg + 1);
+		io.close();
 	}
+
+	//CodeSnip{Kattio}
 }
 ```
 
 </JavaSection>
+<PySection>
+
+```py
+with open('planting.in') as read:
+	field_num = int(read.readline())
+	deg = [0 for _ in range(field_num + 1)]  # 1-indexed
+	for _ in range(field_num - 1):
+		field1, field2 = [int(i) for i in read.readline().split()]
+		deg[field1] += 1
+		deg[field2] += 1
+
+max_deg = max(deg)
+print(max_deg + 1, file=open('planting.out', 'w'))
+```
 
+</PySection>
 </LanguageSection>
 
 ## Solution 2 - DFS
 
 **Time Complexity**: $\mathcal{O}(N)$
 
-This can also be solved using DFS. The solution is less elegant but generates a
-valid assignment of grass types in the $\texttt{ans}$ array.
+This can also be solved using [DFS](/silver/dfs).
+Although it is less elegant, it gets the job done.
```

Comment:
i think the advantage for this is that you've gotten a coloring for all of the fields

Reply:
tru